File: RB101_Programming_Foundations/small_problems/Easy1/Run2/how_many.rb
def count_occurrences(vehicles)
  hash = Hash.new(0)

  vehicles.each do |vehicle|
    hash[vehicle] += 1
  end
  hash
end

vehicles = [
  'car', 'car', 'truck', 'car', 'SUV', 'truck',
  'motorcycle', 'motorcycle', 'car', 'truck'
]

puts count_occurrences(vehicles)

File: js210_fundamentals_of_JS_Programming/problem_sets/small_problems/easy4/how_many.js
function countOccurrences(array) {
  return array.reduce((hash, ele) => {
    hash[ele] = (hash[ele] || 0) + 1;
    return hash;
  }, {});
}

const vehicles = ['car', 'car', 'truck', 'car', 'SUV', 'truck',
  'motorcycle', 'motorcycle', 'car', 'truck'];

console.log(countOccurrences(vehicles));

// console output
// car => 4
// truck => 3
// SUV => 1
// motorcycle => 2
